docs/dev/design/better-doc-urls-handling.rst

```diff
@@ -392,6 +408,95 @@ this allow us to serve subprojects without using a prefix.
      - project
      - The subproject/file doesn't exist
 
+Code example
+------------
+
+This is a simplified version of the implementation.
+
+.. code-block:: python
+
+   from readthedocs.projects.models import Project
+
+   LANGUAGES = {"es", "en"}
+
+
+   def resolve(canonical_project: Project, path_parts: list[str], check_subprojects=True):
+      # Multiversion project.
+      language = path_parts[0]
+      version_slug = path_parts[1]
+      if not canonical_project.single_version and language in LANGUAGES:
+         if canonical_project.language == language:
+               project = canonical_project
+         else:
+               project = canonical_project.translations.filter(language=language).first()
+         if project:
+               version = project.versions.filter(slug=version_slug).first()
+               if version:
+                  return project, version, "/".join(path_parts[2:])
+               return project, None, None
+
+      # Subprojects.
+      prefix = path_parts[0]
+      project_slug = path_parts[1]
+      if check_subprojects and prefix == "projects":
```

Comment:
Yea, I am a little confused here too. I thought the goal of this project was to improve our handling to allow us to be smarter about these cases. It seems this is just changing our logic from using regex to parse the path to using code logic? If that's the only change, I don't see how it's a huge benefit, given that all our other URL's are defined by regex..

Reply:
> @humitos to your point, if we're serving subprojects there will always be some kind of overlap. You can't serve 2 different directories from the same path and not have an order that they are evaluated. I think that's fine, as long as we let users decide that the prefix is, instead of hard-coding projects.

Exactly. That's what I'm saying.

The proposed implementation in this document is hard-coding `/projects/` and that's why I raised this here.

Reply:
> It seemed in the design doc that custom prefixes were supported, is that not shown in this code example?

I thought that was part of the custom URLs part that we wanted to not support yet, I can update the code example to integrate the suffix.